### PR TITLE
Fix file search by file type

### LINKED ISSUE #71: Search for files by file type
As a user, I want to be able to search for files by file type to help me reach the file quicker.

### PR DESCRIPTION
**Pull request by mojeedkusimo**
# Description
PR developed for user to be able to search by any file type specified.

> Fixes #71 

# How Has This Been Tested?
Tested via postman

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no lint warnings

### CODE REVIEW

File: backend/controllers/file.controller.js [hunk 1]
```diff
@@ -140,6 +140,30 @@ exports.getAllDeletedFiles = async (req, res) => {
...
 
+exports.searchByType = async (req, res) => {
+
+  try {
+    const { data } = await File.fetchAll();
+    const { fileType } = req.query;
+
+    if (fileType) {
+      const fileSearch = data.filter((file) => {
+          if (file.name[file.name.length - 4] === '.') {
```
Comment (omzi) [inline backend/controllers/file.controller.js]: Here, you basically assumed ALL file extensions are three characters long by doing this.

File: backend/controllers/file.controller.js [hunk 2]
```diff
@@ -140,6 +140,30 @@ exports.getAllDeletedFiles = async (req, res) => {
...
+              return file.name.slice(file.name.length - 3,) === fileType
+          }
+          return file.name.slice(file.name.length - 4,) === fileType
+      });
+
+      if (fileSearch.length === 0) {
+        return res.status(404).json(`Sorry, there is no file type: ${fileType}, try search with the file extention e.g pdf, mp3, doc, docx, png, jpeg`);
```
Comment (omzi) [inline backend/controllers/file.controller.js]: Then, in this response, you add **jpeg** which is four characters long.

Comment (omzi) [reply backend/controllers/file.controller.js]: Refactor your logic. The `File` schema has the `type` property which you can easily leverage for that.

**Review by omzi — CHANGES_REQUESTED** [review]

**Review by omzi — APPROVED** [review]

**Comment by mojeedkusimo** [comment]
An endpoint that returns the json format result a user searched for by file type

Live link: https://companyfiles.zuri.chat/api/v1/files/searchByType?fileType=mp28

available file types currently on the db include: mp2878, mp6, mp28

**Pull request merged**